s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java

```diff
@@ -0,0 +1,246 @@
+import pl.cyfronet.s4e.ex.BadRequestException;
+import pl.cyfronet.s4e.ex.NotFoundException;
+import pl.cyfronet.s4e.ex.PasswordResetTokenExpiredException;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.temporal.TemporalAmount;
+import java.util.UUID;
+
+import static com.github.npathai.hamcrestopt.OptionalMatchers.isEmpty;
+import static com.github.npathai.hamcrestopt.OptionalMatchers.isPresent;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@BasicTest
+@Slf4j
+public class PasswordServiceTest {
+    private static final String PROFILE_EMAIL = "get@profile.com";
+    private static final TemporalAmount EXPIRE_IN = Duration.ofDays(1);
+
+    @Autowired
+    private TestDbHelper testDbHelper;
+
+    @Autowired
+    private PasswordResetRepository passwordResetRepository;
+
+    @Autowired
+    private AppUserRepository appUserRepository;
+
+    @Autowired
+    private PasswordService passwordService;
+
+    @Autowired
+    private AppUserService appUserService;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    private AppUser appUser;
+
+    @BeforeEach
+    public void setUp() {
+        reset();
+        appUser = appUserRepository.save(AppUser.builder()
+                .email(PROFILE_EMAIL)
+                .name("Get")
+                .surname("Profile")
+                .password(passwordEncoder.encode("password"))
+                .enabled(true)
+                .build());
+    }
+
+    @AfterEach
+    void tearDown() {
+        reset();
+    }
+
+    private void reset() {
+        testDbHelper.clean();
+    }
+
+    @Test
+    public void shouldValidateThatTokenIsNotValid() {
+        PasswordReset token = PasswordReset.builder()
+                .token(UUID.randomUUID().toString())
+                .appUser(appUser)
+                .expiryTimestamp(LocalDateTime.now().minus(EXPIRE_IN))
+                .build();
+        passwordResetRepository.save(token);
+
+        assertThrows(PasswordResetTokenExpiredException.class, () -> passwordService.validate(token.getToken()));
+        assertThrows(NotFoundException.class, () -> passwordService.validate("notFoundToken"));
+    }
+
+    @Test
+    public void shouldValidateThatTokenIsValid() {
+        PasswordReset token = PasswordReset.builder()
+                .token(UUID.randomUUID().toString())
+                .appUser(appUser)
+                .expiryTimestamp(LocalDateTime.now().plus(EXPIRE_IN))
+                .build();
+        passwordResetRepository.save(token);
+        try {
+            passwordService.validate(token.getToken());
+        } catch (NotFoundException e) {
+            fail("Token not found");
+        } catch (PasswordResetTokenExpiredException e) {
+            fail("Token expired");
+        }
```

Comment:
Remove the try-catch here. If it throws, the test will fail anyway. Alternatively, pass the exception as the second argument of the `fail` call.

Reply:
yeah, not sure how that happen :D